lib/tesla.ex

```diff
@@ -97,22 +97,16 @@ defmodule Tesla.Middleware do
 
   ## Writing custom middleware
 
-  Writing custom middleware is as simple as creating a module with `call/3` function that:
-    * (optionally) read and/or writes request data
-    * calls `Tesla.run/2`
-    * (optionally) read and/or writes response data
+  Writing custom middleware is as simple as creating a module implementing `c:Tesla.Middleware.call/3`.
 
-  `call/3` params:
-    * env - `Tesla.Env` struct that stores request/response data
-    * stack - middlewares that should be called after current one
-    * options - middleware options provided by user
+  See `c:Tesla.Middleware.call/3` for details.
 
-  #### Example
+  ### Example
 
       defmodule MyProject.InspectHeadersMiddleware do
         @behaviour Tesla.Middleware
 
-        @impl true
+        @impl Tesla.Middleware
```

Comment:
Note: I haven't test this with older version of Elixir - hopefully CI catches it